doc/read-the-docs-site/adr/0007-observability.rst

```diff
@@ -0,0 +1,63 @@
+.. _adr7:
+
+ADR 7: Marconi observability
+============================
+
+Date:
+
+Authors
+-------
+
+koslambrou <konstantinos.lambrou@iohk.io>
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+Whenever we add new features in the indexers required by ``marconi-sidechain``, we inevitably end up in a state where we introduce a significant regression in sync time and/or memory usage.
+Currently, developers who are adding new features will typically test those features by running ``marconi-sidechain`` on the Cardano preprod network.
+However, that network is very small compared to the Cardano mainnet, and cannot adequatly detect sync time and memory usage regressions.
+Morever, developers are not always in the position to test their changes on mainnet because sync time typically takes around 20h, and they need a computer with at least around 32GB of RAM (~12GB for ``marconi-sidechain`` +  ~12GB for ``cardano-node`` + other processes).
+This large sync times and memory usage are typically because of the ``EpochState`` indexer which uses a large amount of memory because of the ``LedgerState`` computation.
+Even if we manage to remove the ``LedgerState`` computation as planned, sync time is still expected to be hours long.
+
+What we want is a dedicated server which runs the ``marconi-sidechain`` executable daily and exposes different metrics in order to detect any regressions in sync time, CPU, memory and disk usage throught graphical visualizations.
+These graphs would also be used for external reporting as part of our QA practices.
+Then, daily monitoring of ``marconi-sidechain`` will be done by the team, and any detected regressions will be brought to the attention of team before filling a bug.
+
+Decision
+--------
+
+* We will expose Prometheus metrics for our chain-indexers, more specifically, ``marconi-sidechain``.
+  The bare minimum of metrics we should provide are:
+    * CPU, memory and disk usage over time
+    * Total number of processed blocks over time
+  These metrics will be exposed in an HTTP endpoint such as ``/metrics``.
+
+* We will bootstrap a "benchmarking" machine for the team.
+    * Create a private repository named ``marconi-ops``.
+    * Create an AWS instance based of a NixOS image and give the main engineers SSH access.
+    * Save the configuration on version control.
+    * Setup up manual deployment of configuration with ``Morph``.
+    * Setup up continuous deployment of configuration with ``Morph`` for each commit on the ``main`` branch of ``marconi-ops``.
+
+* We will add cardano-node preview, preprod and mainnet services which continuously run in the benchmarking machine.
```

Comment:
Will the cardano-node instances all be running simultaneously to keep fully synced at all times?
If so, that could have an impact on benchmarking results when real-world instances would typically use a single network (unless there is some isolation).

Reply:
But once synced, pre-prod and preview use very little resources, right? I feel that it wouldn't impact the machine that much.

Reply:
Yes perhaps. Let's see :)